docs/data/data-grid/tree-data/tree-data.md

```diff
@@ -83,60 +110,64 @@ If you want to access the grouping column field, for instance, to use it with co
 />
 ```
 
-## Group expansion
+## Group expansion with tree data
 
-Group expansion for tree data works similarly to how it's described in [Row grouping—Group expansion](/x/react-data-grid/row-grouping/#group-expansion).
+For complete details on customizing the group expansion experience, see [Row grouping—Group expansion](/x/react-data-grid/row-grouping/#group-expansion).
+The implementation and behavior are the same when working with tree data.
 
-## Automatic parent and child selection
+## Automatic parent and child selection with tree data
 
-The behavior for automatic parent and child selection is the same as for row grouping—see [the relevant section](/x/react-data-grid/row-grouping/#automatic-parent-and-child-selection) in that feature's documentation for details.
+For complete details on automatic parent and child selection, see [Row grouping—Automatic parent and child selection](/x/react-data-grid/row-grouping/#automatic-parent-and-child-selection).
+The implementation and behavior are the same when working with tree data.
 
 ## Gaps in the tree
 
-If some entries are missing to build the full tree, the Data Grid Pro will automatically create rows to fill those gaps.
+If the tree data provided is missing levels in the hierarchy, the Data Grid Pro will automatically create the rows needed to fill the gaps.
+Consider a simple dataset with two rows:
 
-{{"demo": "TreeDataWithGap.js", "bg": "inline", "defaultCodeOpen": false}}
+```js
+[{ path: ['A'] }, { path: ['A', 'B', 'C'] }];
+```
 
-## Filtering
+This tree data implies a `{ path: ["A", "B"] }` row that's not present.
+Instead of crashing, the Grid creates an internal row so it can render without errors.
```

Comment:
Nitpick:

Maybe, to match the technical naming which is "auto generated row":

```suggestion
Instead of crashing, the Grid generates an internal row so it can render without errors.
```